lib/git/worktree_test.go

```diff
@@ -0,0 +1,60 @@
+package git
+
+import (
+	"os/exec"
+	"testing"
+
+	"github.com/enfabrica/enkit/lib/errdiff"
+
+	"github.com/prashantv/gostub"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewTempWorktree(t *testing.T) {
+	testCases := []struct {
+		desc         string
+		committish   string
+		wantCmds     []*exec.Cmd
+		wantErr      string
+		wantCloseErr string
+	}{
+		{
+			desc:       "successfully creates temp worktree",
+			committish: "some_branch_name",
+			wantCmds: []*exec.Cmd{
+				&exec.Cmd{
+					Path: "git",
+					Args: []string{"worktree", "add", "foo", "some_branch_name"},
+					Dir:  "/foo/bar",
+				},
+				&exec.Cmd{
+					Path: "git",
+					Args: []string{"worktree", "remove", "foo"},
+					Dir:  "/foo/bar",
+				},
+			},
+		},
+	}
+	for _, tc := range testCases {
```

Comment:
minor: the test mocks out git (good!) and checks that the generated commands are as expected (also good!).

But does not test that the actual commands are valid git commands. Is it worth adding a simple test that 1) opens a temp directory (maybe just some testdata/ moved over), and does something like 2) git init, git add . git commit -m "blah", and then 3) tries the worktree commands?

(it's more of an integration test, but should take a few seconds to run)